test/components/extractors/test_llm_metadata_extractor.py

```diff
@@ -145,31 +141,6 @@ def test_to_dict_openai_using_chat_generator(self, monkeypatch):
         }
 
     def test_from_dict_openai(self, monkeypatch):
-        monkeypatch.setenv("OPENAI_API_KEY", "test-api-key")
-        extractor_dict = {
-            "type": "haystack.components.extractors.llm_metadata_extractor.LLMMetadataExtractor",
-            "init_parameters": {
-                "prompt": "some prompt that was used with the LLM {{document.content}}",
-                "expected_keys": ["key1", "key2"],
-                "raise_on_failure": True,
-                "generator_api": "openai",
-                "generator_api_params": {
-                    "api_base_url": None,
-                    "api_key": {"env_vars": ["OPENAI_API_KEY"], "strict": True, "type": "env_var"},
-                    "generation_kwargs": {},
-                    "model": "gpt-4o-mini",
-                    "organization": None,
-                    "streaming_callback": None,
-                },
-            },
-        }
-        extractor = LLMMetadataExtractor.from_dict(extractor_dict)
-        assert extractor.raise_on_failure is True
-        assert extractor.expected_keys == ["key1", "key2"]
-        assert extractor.prompt == "some prompt that was used with the LLM {{document.content}}"
-        assert extractor.generator_api == LLMProvider.OPENAI
```

Comment:
We should keep this test and just change this line to 
`assert isinstance(extractor._chat_generator, OpenAIChatGenerator)`

Reply:
done in https://github.com/deepset-ai/haystack/pull/9099/commits/f026f52f9b7cbcecad4f2a90e3a1ba9f121e82bc